zulipterminal/ui_tools/boxes.py

```diff
@@ -476,14 +476,11 @@ def soup2markup(self, soup: Any, **state: Any) -> List[Any]:
                     # Prepend org url to convert it to an absolute link
                     link = urljoin(self.model.server_url, link)
 
-                if link == text:
-                    # If the link and text are same
-                    # usually the case when user just pastes
-                    # a link then just display the link
-                    markup.append(('msg_link', text))
-                else:
-                    markup.append(
-                        ('msg_link', '[' + text + ']' + '(' + link + ')'))
+                text = text if text else link
+                markup.append(
+                    ('msg_link', '{} [{}]'.format(text, self.link_index))
```

Comment:
Currently, I am showing link as text whenever text is not present. This, however, could be noisy when the link is comparatively long.

Reply:
Having duplicate content seems confusing and noisy, yes.

Reply:
Yeah, I'd prefer only having one when there's no text.

Reply:
As per the recent discussion, the inclination is towards only using the link in either case.